sae.c

```diff
@@ -444,14 +458,14 @@ process_confirm (struct candidate *peer, struct ieee80211_mgmt_frame *frame, int
     offset = BN_num_bytes(peer->grp_def->prime) - BN_num_bytes(x);
     memset(tmp, 0, offset);
     BN_bn2bin(x, tmp + offset);
-    CN_Update(&ctx, tmp, BN_num_bytes(peer->grp_def->prime));
+    CN_Update(ctx, tmp, BN_num_bytes(peer->grp_def->prime));
     offset = BN_num_bytes(peer->grp_def->prime) - BN_num_bytes(y);
     memset(tmp, 0, offset);
     BN_bn2bin(y, tmp + offset);
-    CN_Update(&ctx, tmp, BN_num_bytes(peer->grp_def->prime));
+    CN_Update(ctx, tmp, BN_num_bytes(peer->grp_def->prime));
 
-    CN_Final(&ctx, tmp);
-    HMAC_CTX_cleanup(&ctx);
+    CN_Final(ctx, tmp);
+    HMAC_CTX_free(ctx);
```

Comment:
I think this should go to out, otherwise there are leaks on the error paths

Reply:
will fix